pkg/client/client.go

```diff
@@ -24,8 +24,11 @@ import (
 // Config returns a *rest.Config, using either the kubeconfig (if specified) or an in-cluster
 // configuration.
 func Config(kubeconfig string) (*rest.Config, error) {
-	if len(kubeconfig) > 0 {
-		return clientcmd.BuildConfigFromFlags("", kubeconfig)
+	loader := clientcmd.NewDefaultClientConfigLoadingRules()
+	loader.ExplicitPath = kubeconfig
+	clientConfig, err := clientcmd.BuildConfigFromKubeconfigGetter("", loader.Load)
```

Comment:
There are very few places in the kube code base that use `BuildConfigFromKubeconfigGetter`. It might be better to match more closely what's in https://github.com/kubernetes/kubernetes/blob/775f5d232d21eccc967a2007e5290f04ef15d93b/pkg/kubectl/cmd/util/factory_client_access.go#L156-L174, but changing it up a bit:

```go
return clientcmd.NewNonInteractiveDeferredLoadingClientConfig(loader, nil), nil
```

We may want to have that `nil` be actual overrides.

Reply:
That can work as well.  Is your concern that since it's not used much the behavior may change over time in an unanticipated way and that we are reliant upon the implementation details?

Reply:
It was more that "kubectl does it a different way" 😄 . The kube client config loading code is difficult for me to keep all of it in my head at the same time... I did test your branch locally to confirm that it still loaded the in-cluster config (when running as a pod), and that did work. The more that I think about it, I'm fine with this.

Reply:
Sounds good.  I dug in a bit deeper to the code sample you gave and the main things I think differ are around the overrides that you mentioned.  If we go with this for now and our needs change, we can certainly just shift to a more explicit resolution process as your code sample gives.  

I agree that it's a lot to grok and I was surprised there wasn't something a bit more "direct" in the client package that others could just use to keep ti consistent